catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/SceneTransitionWithSoundBrickStageTest.java

```diff
@@ -0,0 +1,167 @@
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.PlaySoundBrick;
+import org.catrobat.catroid.content.bricks.SceneTransitionBrick;
+import org.catrobat.catroid.content.bricks.WaitBrick;
+import org.catrobat.catroid.io.ResourceImporter;
+import org.catrobat.catroid.io.SoundManager;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.uiespresso.stage.utils.ScriptEvaluationGateBrick;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+import static org.catrobat.catroid.common.Constants.SOUND_DIRECTORY_NAME;
+
+@RunWith(AndroidJUnit4.class)
+public class SceneTransitionWithSoundBrickStageTest {
+
+	private ScriptEvaluationGateBrick lastBrickFirstScript;
+	private ScriptEvaluationGateBrick lastBrickSecondScript;
+
+	private String firstSceneName;
+	private Script secondScript;
+
+	private static ProjectManager projectManager;
+	private File soundFile;
+
+	@Rule
+	public BaseActivityTestRule<StageActivity> baseActivityTestRule = new
+			BaseActivityTestRule<>(StageActivity.class, false,
+			false);
+
+	@BeforeClass
+	public static void setUpProjectManager() {
+		projectManager = ProjectManager.getInstance();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(getClass().getSimpleName());
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() throws IOException {
+		TestUtils.deleteProjects(getClass().getSimpleName());
+		if (soundFile != null && soundFile.exists()) {
+			soundFile.delete();
+		}
+	}
+
+	@Test
+	public void testStopSoundOnSceneTransition() {
+		runProject(lastBrickSecondScript);
+		assertFalse(getMediaplayer().isPlaying());
+	}
+
+	@Test
+	public void testContinueSoundAfterSceneTransition() {
+		secondScript.addBrick(new SceneTransitionBrick(firstSceneName));
+		runProject(lastBrickFirstScript);
+		assertTrue(getMediaplayer().isPlaying());
+	}
+
+	@Test
+	public void testContinueSoundDoesNotStartFromBeginning() {
+		secondScript.addBrick(new SceneTransitionBrick(firstSceneName));
+		runProject(lastBrickFirstScript);
+		MediaPlayer mediaPlayer = getMediaplayer();
+		assertTrue(mediaPlayer.isPlaying());
+		assertTrue(mediaPlayer.getCurrentPosition() > 0);
```

Comment:
PLEASE EXPLAIN: Have you checked this test in test-first style, i.e., before you implemented the fix? And what is the unit for the current position here? Can't this asset be true even if the sound was restarted, because of timing issues? Please clarify and confirm.

Reply:
I did start my tests without my fix and all of them failed. 
The current position is the time played in milliseconds.
Yes, I think timing issues are a potential problem here, but I did not find a better way to check this.

Reply:
Thanks! How about, after starting the sound in the first scene, to wait for 0.1 second while still in the first scene using a "wait 0.1 seconds" brick, and then when returning back from the second scene immediately checking that the current position is > 100? Would that work?

Reply:
Oh I forgot to mention that, I already have a "wait 0.5 seconds" brick and after some testing it seems that I have to wait at least 0.3 seconds for the sound to start.
This seems to be very inconsistent and unfortunately I don't think a more precise check is possible.

Reply:
Well, shouldn't it then be >200? You could compare to > 150 instead of > 0, right?

Reply:
I just want to make these tests future proof (Moore's law...)

Reply:
After running the test three times the position of the Sound when transitioning back were: 227, 82, 141 ms
It really varies a lot.

Reply:
> 50?

Reply:
Thanks!